File: output/objc/PersonBuilder.m
#import "PersonBuilder.h"

@implementation PersonBuilder
{
    NSString* _firstName;
    NSString* _lastName;
    NSString* _nickName;
    NSInteger _age;
    BOOL _canOrder;
    Address* _address;
    NSArray<Address*>* _all;
}

- (Person *)buildPerson
{
    return [Person personWithFirstName:_firstName lastName:_lastName nickName:_nickName age:_age canOrder:_canOrder address:_address all:_all];
}

#pragma mark - Initializers

+ (instancetype)builder
{
    return [self new];
}

+ (instancetype)builderWithPerson:(Person *)existingPerson
{
    PersonBuilder* builder = [self builder];
    builder = [builder withFirstName:existingPerson.firstName];
    builder = [builder withLastName:existingPerson.lastName];
    builder = [builder withNickName:existingPerson.nickName];
    builder = [builder withAge:existingPerson.age];
    builder = [builder withCanOrder:existingPerson.canOrder];
    builder = [builder withAddress:existingPerson.address];
    builder = [builder withAll:existingPerson.all];
    return builder;
}

#pragma mark - Property setters

- (instancetype)withFirstName:(NSString*)firstName
{
    _firstName = [firstName copy];
    return self;
}

- (instancetype)withLastName:(NSString*)lastName
{
    _lastName = [lastName copy];
    return self;
}

- (instancetype)withNickName:(NSString*)nickName
{
    _nickName = [nickName copy];
    return self;
}

- (instancetype)withAge:(NSInteger)age
{
    _age = age;
    return self;
}

- (instancetype)withCanOrder:(BOOL)canOrder
{
    _canOrder = canOrder;
    return self;
}

- (instancetype)withAddress:(Address*)address
{
    _address = [address copy];
    return self;
}

- (instancetype)withAll:(NSArray<Address*>*)all
{
    _all = [all copy];
    return self;
}

@end

File: output/swift/PersonBuilder.swift
import Foundation

struct PersonBuilder {
	var firstName: String
	var lastName: String
	var nickName: String
	var age: Int
	var address: Address
    
    func build () -> Person {
        return Person(firstName: firstName, lastName: lastName, nickName: nickName, age: age, address: address)
    }
    
    //MARK: Initializers
    
    init(existingPerson: Person) {
        firstName = existingPerson.firstName
        lastName = existingPerson.lastName
        nickName = existingPerson.nickName
        age = existingPerson.age
        address = existingPerson.address
    }
    
    //MARK: Property setters
    
    mutating func withFirstName(_ firstName: String) -> PersonBuilder {
        self.firstName = firstName
        return self
    }

    mutating func withLastName(_ lastName: String) -> PersonBuilder {
        self.lastName = lastName
        return self
    }

    mutating func withNickName(_ nickName: String) -> PersonBuilder {
        self.nickName = nickName
        return self
    }

    mutating func withAge(_ age: Int) -> PersonBuilder {
        self.age = age
        return self
    }

    mutating func withAddress(_ address: Address) -> PersonBuilder {
        self.address = address
        return self
    }
}
